src/java.base/share/classes/java/lang/classfile/ClassHierarchyResolver.java

```diff
@@ -82,6 +83,9 @@ sealed interface ClassHierarchyInfo permits ClassHierarchyImpl.ClassHierarchyInf
          * @return the info indicating the super class
          */
         static ClassHierarchyInfo ofClass(ClassDesc superClass) {
+            //todo: uncomment later
+            // this causes CorpusTest to fail
+            // requireNonNull(superClass);
```

Comment:
This is valid, as the super class descriptor of `java.lang.Object` is `null` (it’s the only one allowed and required to extend `null`).
```suggestion
```